command/apply.go

```diff
@@ -24,141 +24,136 @@ var applyCommand = &cobra.Command{
-				confText, err := ioutil.ReadFile("./ticket_fields" + configExtension)
-				if err != nil {
-					exe.Error(err.Error())
-					return err
-				}
-				conf.TicketFields, err = parser.ParseTicketFields(string(confText))
-				if err != nil {
-					exe.Error(err.Error())
-					return err
-				}
-				exe.Success("OK")
-				return nil
-			})
+	// Create parser according to config extension
+	configExtension := "." + configFormat
+	parser, err := config.NewParserFromExtension(configExtension)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
 
-			exe.Step("Loading ticket_forms"+configExtension, func() error {
-				confText, err := ioutil.ReadFile("./ticket_forms" + configExtension)
-				if err != nil {
-					exe.Error(err.Error())
-					return err
-				}
-				conf.TicketForms, err = parser.ParseTicketForms(string(confText))
-				if err != nil {
-					exe.Error(err.Error())
-					return err
-				}
-				exe.Success("OK")
-				return nil
-			})
+	//TODO: check if all configs exists
+	exe.Step("Detecting config files", func() error {
+		exe.Success("Found (number of files) files")
+		return nil
+	})
 
-			exe.Step("Loading triggers"+configExtension, func() error {
-				confText, err := ioutil.ReadFile("./triggers" + configExtension)
-				if err != nil {
-					exe.Error(err.Error())
-					return err
-				}
-				conf.Triggers, err = parser.ParseTriggers(string(confText))
-				if err != nil {
-					exe.Error(err.Error())
-					return err
-				}
-				exe.Success("OK")
-				return nil
-			})
+	exe.Step("Loading configuration files", func() error {
+		exe.Step("Loading ticket_fields"+configExtension, func() error {
+			confText, err := ioutil.ReadFile("./ticket_fields" + configExtension)
+			if err != nil {
+				exe.Error(err.Error())
+				return err
+			}
+			conf.TicketFields, err = parser.ParseTicketFields(string(confText))
+			if err != nil {
+				exe.Error(err.Error())
+				return err
+			}
+			exe.Success("OK")
 			return nil
 		})
 
-		exe.Step("Applying patch to Zendesk", func() error {
-			if err := exe.StepIf(len(conf.TicketFields) > 0, "Creating ticket fields...", stepCreateTicketFields(exe, zd, conf, currentState)); err != nil {
-				return err
-			}
-			if err := exe.StepIf(len(conf.TicketForms) > 0, "Creating ticket forms...", stepCreateTicketForms(exe, zd, conf, currentState)); err != nil {
+		exe.Step("Loading ticket_forms"+configExtension, func() error {
+			confText, err := ioutil.ReadFile("./ticket_forms" + configExtension)
+			if err != nil {
+				exe.Error(err.Error())
 				return err
 			}
-			if err := exe.StepIf(len(conf.Triggers) > 0, "Creating triggers...", stepCreateTriggers(exe, zd, conf, currentState)); err != nil {
+			conf.TicketForms, err = parser.ParseTicketForms(string(confText))
+			if err != nil {
+				exe.Error(err.Error())
 				return err
 			}
+			exe.Success("OK")
 			return nil
 		})
-		// TODO:
-		// - make request
-		// - request Zendesk API
-		// - parse response
```

Comment:
Are these TODO are all solved?

Reply:
Yes, these features have been implemented in each step, stepCreateTicketFields, stepCreateTicketForms and stepCreateTriggers.

Reply:
👍